### ISSUE TITLE
[Bug]: aws_route53_vpc_association_authorization InvalidPaginationToken during terraform plan

### ISSUE THREAD

**Issue by bFekete** [hashicorp/terraform-provider-aws#40585]
### Terraform Core Version

1.6.6

### AWS Provider Version

5.66.0,5.67.0,5.68.0,5.69.0,5.70.0,5.71.0,5.72.0,5.73.0,5.74.0,5.75.0,5.76.0,5.77.0,5.78.0,5.79.0,5.80.0,5.81.0

### Affected Resource(s)

* aws_route53_vpc_association_authorization

### Expected Behavior

Terraform plan should succeed and determine whether or not to make changes.

### Actual Behavior

Terraform plan fails with an `InvalidPaginationToken` error on `ListVPCAssociationAuthorizations` when running multiple terraform workspaces concurrently. Using a central network account in a micro account architecture which is why we have multiple workspaces running concurrently with a provider to the central network account.

I believe what is happening, the resource isn't properly retrying and handling a 400 Bad Request, Throttling error.

https://github.com/hashicorp/terraform-provider-aws/blob/c45a7d8e90ea23fdf52fab8a9a692d2754c37d18/internal/service/route53/vpc_association_authorization.go#L189

> All Amazon Route 53 API requests
For the [Amazon Route 53 APIs](https://docs.aws.amazon.com/Route53/latest/APIReference/API_Operations_Amazon_Route_53.html) five requests per second per AWS account. If you submit more than five requests per second, Amazon Route 53 returns an HTTP 400 error (Bad request). The response header also includes a Code element with a value of Throttling and a Message element with a value of Rate exceeded.

### Relevant Error/Panic Output Snippet

```shell
> Error: reading Route53 VPC Association Authorization (<zone_id>:<vpc_id>): operation error Route 53: ListVPCAssociationAuthorizations, https response error StatusCode: 400, RequestID: <request_id>, InvalidPaginationToken: <long_string_here>
```

### Terraform Configuration Files

```hcl
resource "aws_route53_vpc_association_authorization" "this" {
  provider = aws.network-hub

  for_each = local.zone_ids_json
  zone_id  = each.key
  vpc_id    = aws_vpc.this.id
}
```

### Steps to Reproduce

1. Cause a r53 throttling error. simple bash + curl command
2. Run terraform 

### Important Factoids

Using AFT in a micro account architecture. Running many customization workspaces when communicating to a single central network aws account.

Route53 API has a limit of five requests per second.

### References

https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DNSLimitations.html

https://github.com/hashicorp/terraform-provider-aws/blob/c45a7d8e90ea23fdf52fab8a9a692d2754c37d18/internal/service/route53/vpc_association_authorization.go#L189

### Would you like to implement a fix?

No

**Comment by mrg2k8** [comment]
A workaround for this specific use case is to use Route 53 Profiles, but they're not free.

**Comment by MinhHungPhan** [comment]
@mrg2k8  I agree with you. I spoke with our TAM yesterday, and they recommended using Route 53 Profiles, but the cost is prohibitively high 😅
